### ISSUE TITLE
Training from pre-trained weights fails for Single Class

### ISSUE THREAD

**Issue by fho306** [ultralytics/yolov5#594]
The following bug appears when training yolov5s on a single class dataset or with the --single-cls flag.
```
File "train.py", line 461, in <module>
    train(hyp, tb_writer, opt, device)
  File "train.py", line 284, in train
    loss, loss_items = compute_loss(pred, targets.to(device), model)  # scaled by batch_size
  File "/content/yolov5/utils/utils.py", line 496, in compute_loss
    lcls *= h['cls'] * s
RuntimeError: a leaf Variable that requires grad is being used in an in-place operation.
```
Also tested in the official Colab notebook.

**Comment by fho306** [comment]
To reproduce, follow the steps in the official colab normaly, then start training with ` --single-cls ` flag or change `nc: 1` in the `data.yaml`

**Comment by fho306** [comment]
It seems that the issue only exists when using the pretrained models. Commenting out the mentioned line seemed ok for me, since class loss is irrelevant for single class data.

**Comment by glenn-jocher** [comment]
@fho306 thanks for finding this bug! Coincidentally I just resolved this error in #593. Please git pull and try again.

**Comment by fho306** [comment]
@glenn-jocher Fixed! Thank you very much for your work.
